pkg/policy/api/l4.go

```diff
@@ -55,6 +55,36 @@ func (p PortProtocol) Covers(other PortProtocol) bool {
 	return true
 }
 
+// K8sSecret is a reference to a k8s secret.
+type K8sSecret struct {
+	// Namespace is the k8s namespace in which the secret exists. If
+	// namespace is omitted, the namespace of the enclosing rule is assumed.
+	Namespace string `json:"namespace,omitempty"`
+
+	// Name is the name of the k8s secret
+	Name string `json:"name,omitempty"`
+}
+
+// TLSContext provides TLS configuration via reference to k8s secrets.
+type TLSContext struct {
+	// Certificate is a chain of certificates presented to the remote party.
+	// If specified for an originating TLS context, then this is used as a
+	// client certificate
+	Certificate K8sSecret `json:"certificate,omitempty"`
```

Comment:
@jrajahalme is the intention for these to be consumable only as k8s secrets? or would it also be possible to mount cert material to be used as `TLSContext`

Reply:
> @jrajahalme is the intention for these to be consumable only as k8s secrets?

yes

> or would it also be possible to mount cert material to be used as `TLSContext`

@willdeuschle where how/would you mount it into cilium?

Reply:
@aanm Our configuration would be static enough such that we could deploy the agent with the certs mounted as volumes. This isn't a hard and fast requirement, it just seems like it would be lower lift.